pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -398,6 +404,48 @@ func (h *VMController) SetHaltIfInsufficientResourceQuota(_ string, vm *kubevirt
 	return vm, h.stopVM(vm, err.Error())
 }
 
+// RestartFromPausedIOError automatically restarts VMs from status flipping between Paused and Running
+// As a workaround for the issue https://github.com/harvester/harvester/issues/4633
+func (h *VMController) RestartFromPausedIOError(_ string, vm *kubevirtv1.VirtualMachine) (*kubevirtv1.VirtualMachine, error) {
+	if vm == nil || vm.DeletionTimestamp != nil || vm.Spec.Template == nil {
+		return vm, nil
+	}
+
+	// if the vm is not paused or it's under restarting, skip it.
+	if vm.Status.PrintableStatus != kubevirtv1.VirtualMachineStatusPaused || len(vm.Status.StateChangeRequests) != 0 {
+		return vm, nil
+	}
+
+	vmi, err := h.vmiCache.Get(vm.Namespace, vm.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, volume := range vmi.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		if ready, err := h.isPVCReady(vm.Namespace, volume.PersistentVolumeClaim.ClaimName); err != nil {
+			return nil, err
+		} else if !ready {
+			logrus.Debugf("Volume %s is not healthy in VM %s/%s, wait for 5 seconds to check VM again", volume.Name, vm.Namespace, vm.Name)
+			h.vmController.EnqueueAfter(vm.Namespace, vm.Name, 5*time.Second)
+			return vm, nil
+		}
+	}
+
+	for _, condition := range vmi.Status.Conditions {
+		if condition.Type == kubevirtv1.VirtualMachineInstancePaused && condition.Status == corev1.ConditionTrue &&
+			condition.Reason == "PausedIOError" {
```

Comment:
Is `PausedIOError` defined anywhere? let's refer it or define it. Reuse also in next line log.

Reply:
No, I traced KubeVirt source code. They don't define `PausedIOError`.

https://github.com/kubevirt/kubevirt/blob/85e9173c308434a6e2a9099f89eb631782e67bde/pkg/virt-handler/vm.go#L1465

Reply:
Then let's define `PausedIOError` as a const in this go file.